pkg/kube/kubehttp.go

```diff
@@ -376,6 +390,91 @@ func (k *KubeHTTP) SetResource(ctx context.Context, resource Resource) error {
 	return nil
 }
 
+func (k *KubeHTTP) SetWegoConfig(ctx context.Context, config WegoConfig, namespace string) error {
+	configBytes, err := kyaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed marshalling wego config: %w", err)
+	}
+
+	name := types.NamespacedName{Name: WegoConfigMapName, Namespace: namespace}
+
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+		},
+	}
+
+	if err := k.Client.Get(ctx, name, cm); err != nil {
+		if apierrors.IsNotFound(err) {
+			cm.Data = map[string]string{
+				"config": string(configBytes),
+			}
+
+			if err = k.Client.Create(ctx, cm); err != nil {
+				return fmt.Errorf("failed creating weave-gitops configmap: %w", err)
+			}
+
+			return nil
+		}
+
+		return fmt.Errorf("failed getting weave-gitops configmap: %w", err)
+	}
+
+	cm.Data["config"] = string(configBytes)
+
+	return k.Client.Update(ctx, cm)
+}
+
+func (k *KubeHTTP) GetWegoConfig(ctx context.Context, namespace string) (*WegoConfig, error) {
+	name := types.NamespacedName{Name: WegoConfigMapName, Namespace: namespace}
+
+	cm := &corev1.ConfigMap{}
+
+	if namespace != "" {
+		if err := k.Client.Get(ctx, name, cm); err != nil {
+			if apierrors.IsNotFound(err) {
+				return &WegoConfig{}, ErrWegoConfigNotFound
+			}
+
+			return nil, fmt.Errorf("failed getting weave-gitops configmap: %w", err)
+		}
+	} else {
+		configMap, err := k.getWegoConfigMapFromAllNamespaces(ctx)
+		if err != nil {
+			return &WegoConfig{}, err
+		}
+
+		cm = configMap
+	}
+
+	wegoConfig := &WegoConfig{}
+	if err := kyaml.Unmarshal([]byte(cm.Data["config"]), wegoConfig); err != nil {
+		return nil, nil
```

Comment:
Any particular reason we are not returning this error to the caller?

Reply:
good catch, it should